### PR TITLE
Rework ci and scripts.

### PR DESCRIPTION
**Pull request by neumantm**
- Switch to github actions
 - Move and rename scripts to create some order
 - Change how version bumps work to make it more understandable and less likley to fail in a bad way.
 - Generally improve scripts
 - Remove unneccesary scripts

### CODE REVIEW

File: scriptsAndTools/README.md [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 So use `$ ./scriptsAndTools/<script>.sh`
 
 Most scripts will fail if an argument is missing
 
+If there are any unexpected maven errors, try `mvn clean install` then run the script again. 
+
+
 ## Scripts to be used by developers
-Don't use the other scripts, except if you know what you are doing.
+Only use the other scripts if you know what you are doing.
```
Comment (buehlefs) [inline scriptsAndTools/README.md:12]: What is meant by other scripts? There is only one list of scripts following without any distinction. This only discourages developers from using the scripts but we want them to use some of the scripts before pushing a commit to check formatting etc.

Document bash requirement for scripts!

File: scriptsAndTools/README.md [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
-| `verifyFormat.sh` | Checks all format requirements. `mvn clean install` before using it.
-| `fixSourceFileHeaders.sh` | Replaces the first part of each java file in the repo with the correctHeader.
-| `formatPoms.sh` | Formats the poms according to the requirements.
-| `versionBump.sh` | Initiate a version bump and release.
+| `checkFormat.sh` | Checks all basic format requirements. Does not check javadoc validity (use `mvn javadoc:javadoc` for that)
+| `checkjavaFileComments.sh` | Checks the file comments of all java files. Included in `checkFormat.sh`
+| `checkPomFormat.sh` | Checks the format of the poms. Included in `checkFormat.sh`
+| `fixJavaFileComments.sh` | Replaces the first part of each java file in the repo with the correctHeader.
+| `fixPomFormat.sh` | Formats the poms according to the requirements.
+| `prepareRelease.sh` | Prepares a new release by increasing the version, tagging and increasing the version to the next snapshot-version.
```
Comment (buehlefs) [inline scriptsAndTools/README.md:21]: Maybe add a column for script safety.
`checkFormat.sh` should have no consequences -> safe to execute every time
`fixJavaFileComments.sh` edits (java) files -> safe to execute on clean git state
`prepareRelease.sh` makes commits on extra branch -> double check before executing/pushing branch

Also add a code block with example invocations of the scripts below the table

File: scriptsAndTools/checkFormat.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+if [ ! "$1" == "-v" ] && [ $# -ne 0 ] ; then
+  fail "No argument expected (or single -v)" 2
+fi
+
+if ! which git >/dev/null 2>&1 ;then
+  fail "Need program git" 3
+fi
+
+if ! which mvn >/dev/null 2>&1 ;then
```
Comment (buehlefs) [inline scriptsAndTools/checkFormat.sh:29]: We have the maven wrapper (`mvnw`) configured in the repo. Should we use it consistently for all our scripts?

File: scriptsAndTools/README.md [hunk 4]
```diff
@@ -1,21 +1,74 @@
 # Scripts and Tools
 
-Many of these scripts expect to be run from the root directory of this repository.
-So use `$ ./scriptsAndTools/<script>.sh`
+All of the scripts expect to be run from the root directory of this repository. \
+ **So always cd to the git root first and use `./scriptsAndTools/<script>.sh`!**
 
-Most scripts will fail if an argument is missing
+All non-internal scripts will fail if an requirement or argument is missing
```
Comment (buehlefs) [inline scriptsAndTools/README.md:6]:
```suggestion
All non-internal scripts will fail if a requirement or argument is missing
```

File: scriptsAndTools/README.md [hunk 5]
```diff
@@ -1,21 +1,74 @@
...
 
-If there are any unexpected maven errors, try `mvn clean install` then run the script again. 
+If there are any unexpected maven errors, try `mvn clean install`  and then run the script again.
```
Comment (buehlefs) [inline scriptsAndTools/README.md:8]: This should also use the maven wrapper. Also explain maven wrapper in the repository readme.

File: scriptsAndTools/README.md [hunk 6]
```diff
@@ -1,21 +1,74 @@
...
-| `checkPomFormat.sh` | Checks the format of the poms. Included in `checkFormat.sh`
-| `fixJavaFileComments.sh` | Replaces the first part of each java file in the repo with the correctHeader.
-| `fixPomFormat.sh` | Formats the poms according to the requirements.
-| `prepareRelease.sh` | Prepares a new release by increasing the version, tagging and increasing the version to the next snapshot-version.
+You can use the following scripts during your development.
+Only use the scripts in `internalOrCiOnly` if you know what you are doing.
+
+| Name | Description | Requirements | Arguments | Script Saftey
+| --- | --- | --- |  --- | --- |
+| `checkFormat.sh` | Checks all basic format requirements. Does not check javadoc validity (use `mvn javadoc:javadoc` for that) | `xmlstarlet` and `diff` | none or single `-v` | read only
```
Comment (buehlefs) [inline scriptsAndTools/README.md:17]: consistently use maven wrapper in the readme (or explain it once that every instance of mvn in the readme can be replaced by a call of the maven wrapper).

File: scriptsAndTools/README.md [hunk 7]
```diff
@@ -1,21 +1,74 @@
...
+| `checkJavaFileComments.sh` | Checks the file comments of all java files. Included in `checkFormat.sh` | none | none | read only
+| `checkPomFormat.sh` | Checks the format of the poms. Included in `checkFormat.sh` | `xmlstarlet` and `diff` | none | writes temporary files in working dir and deletes them. Uses existing file names with `.2` appended for the tmp files.
+| `fixJavaFileComments.sh` | Replaces the first part of each java file in the repo with the correctHeader. | none | none | writes to working dir
+| `fixPomFormat.sh` | Formats the poms according to the requirements. | `xmlstarlet` | none | writes to working dir
```
Comment (buehlefs) [inline scriptsAndTools/README.md:21]: Maybe change "to the requirements" -> "to be processable by the scripts" (more specific than unknown requirements)

File: scriptsAndTools/README.md [hunk 8]
```diff
@@ -1,21 +1,74 @@
...
+| `prepareRelease.sh` | Prepares a new release by increasing the version, tagging and increasing the version to the next snapshot-version. | `xmlstarlet` and `git`; no uncommited changes, beeing on an up to
+date master branch; being able to interact with the script | none | writes to working dir and commits. Optionally pushes and deletes branch.
+
+## Examples
+
+Checking the format:
+```
```
Comment (buehlefs) [inline scriptsAndTools/README.md:28]: you could try to specify "bash" or "sh" as the language for the code blocks

File: scriptsAndTools/README.md [hunk 9]
```diff
@@ -1,21 +1,74 @@
...
+ 4. Run the script `prepareRelease` (see example above)
+    1. If you didn't choose to autmatically push, push the new branch and the tags
+ 5. Create a pull request, have someone review it and merge the pull request
+ 6. Create a Github Release from the new tag.
+ 7. Wait for the Github Worflow to run.
+
+### Explanation
+The script changes the version in all poms to the version to release. Then it commits and tags this state.
+Then it changes the version in all poms again. This time to the next version with the `-Snapshot` suffix. 
+This way the development can safley continue afterwards. This is also committed.
```
Comment (buehlefs) [inline scriptsAndTools/README.md:73]:
```suggestion
This way the development can safeley continue afterwards. This is also committed.
```

File: scriptsAndTools/README.md [hunk 10]
```diff
@@ -1,21 +1,74 @@
...
-
-| Name | Description |
-| --- | --- |
-| `checkFormat.sh` | Checks all basic format requirements. Does not check javadoc validity (use `mvn javadoc:javadoc` for that)
-| `checkjavaFileComments.sh` | Checks the file comments of all java files. Included in `checkFormat.sh`
-| `checkPomFormat.sh` | Checks the format of the poms. Included in `checkFormat.sh`
-| `fixJavaFileComments.sh` | Replaces the first part of each java file in the repo with the correctHeader.
-| `fixPomFormat.sh` | Formats the poms according to the requirements.
-| `prepareRelease.sh` | Prepares a new release by increasing the version, tagging and increasing the version to the next snapshot-version.
+You can use the following scripts during your development.
```
Comment (buehlefs) [inline scriptsAndTools/README.md:12]: Document which shells these scripts were tested for.

File: scriptsAndTools/README.md (hunk 7 above)
Comment (neumantm) [reply scriptsAndTools/README.md:21]: This is actually not correct. Our tools don't care about the format. Will change to `Format poms to match the correct format used for poms by this project`

**Pull request merged**